src/gululog_topic.erl

```diff
@@ -126,6 +128,34 @@ truncate(#topic{dir = Dir, idx = Idx, cur = Cur} = Topic, LogId, BackupDir) ->
        IdxFileOpList ++ SegFileOpList}
   end.
 
+%% @doc Delete oldest segment
+%% return the segid that is deleted, return 'false' in case:
+%% 1. nothing to delete
+%% 2. the oldest is also the latest
+%% @end
+-spec delete_oldest_seg(topic()) -> {false | segid(), topic()}.
```

Comment:
return [file_op()] instead.
REASON: this is mostly for higher level callers to log events etc. we can perhaps log it (using lager for example) here It's probably a bad idea to add a logging app dependency in gululog level.